data_model/model.py

```diff
@@ -172,22 +171,25 @@ def get_device_by_id(self, device_id):
     def add_device_to_house(self, house, device_id):
         house += self.devices[device_id]
 
-    def get_devices_for_house(self, house):
-        devs = []
-        for elem in house:
-            if elem in self.devices:
-                devs += elem
-        return devs
+    def get_devices_for_house(self, devices, house_id):
+        for device in devices:
+            if device.house_id == house_id:
+                return device
 
     def add_device_to_room(self, room, device_id):
-        room += device_id
+        room += self.devices[device_id]
+
+    def get_devices_for_room(self, devices, room_id):
+        for device in devices:
+            if device.room_id == room_id:
+                return device
 
 
 class DeviceGroups(Device):
     def __init__(self, device_id):
         self.device_id = device_id
```

Comment:
You probably need multiple device ids in 1 group. In addition to that, you should probably add a device_group_id.